I am getting this error when trying to `doc` or `publishLocal` in SBT. Strangely, this does not occur when compiling or running tests, so I assume that this is related to scaladoc, but am not certain.

As I'm not getting any line numbers I can't determine what is causing this to fail, and can not provide a minimal example to reproduce (the codebase is from work, so I can't release the whole thing, but I may be able to provide a snippet if I can identify what's causing it).

Scala 2.12.6, SBT 1.1.6.

The stack trace follows:

>  sbt:mongosaur> publishLocal
>  [info] Writing Ivy file /home/luciano/src/mongosaur/target/scala-2.12/resolution-cache/com.dinogroup/mongosaur_2.12/0.1.0-SNAPSHOT/resolved.xml.xml
>  [info] Wrote /home/luciano/src/mongosaur/target/scala-2.12/mongosaur_2.12-0.1.0-SNAPSHOT.pom
>  [info] Packaging /home/luciano/src/mongosaur/target/scala-2.12/mongosaur_2.12-0.1.0-SNAPSHOT-sources.jar ...
>  [info] Done packaging.
>  [info] Main Scala API documentation to /home/luciano/src/mongosaur/target/scala-2.12/api...
>  [info] Compiling 16 Scala sources to /home/luciano/src/mongosaur/target/scala-2.12/classes ...
>  model contains 75 documentable templates
>  [info] Done compiling.
>  [info] Packaging /home/luciano/src/mongosaur/target/scala-2.12/mongosaur_2.12-0.1.0-SNAPSHOT.jar ...
>  [info] Done packaging.
>  [error] java.lang.UnsupportedOperationException: Position.start on NoPosition
>  [error] 	at scala.reflect.internal.util.Position.fail(Position.scala:17)
>  [error] 	at scala.reflect.internal.util.UndefinedPosition.start(Position.scala:94)
>  [error] 	at scala.reflect.internal.util.UndefinedPosition.start(Position.scala:90)
>  [error] 	at scala.tools.nsc.ast.DocComments.expandInternal$1(DocComments.scala:356)
>  [error] 	at scala.tools.nsc.ast.DocComments.expandVariables(DocComments.scala:371)
>  [error] 	at scala.tools.nsc.ast.DocComments.expandVariables$(DocComments.scala:319)
>  [error] 	at scala.tools.nsc.Global.expandVariables(Global.scala:36)
>  [error] 	at scala.tools.nsc.ast.DocComments.expandedDocComment(DocComments.scala:102)
>  [error] 	at scala.tools.nsc.ast.DocComments.expandedDocComment$(DocComments.scala:98)
>  [error] 	at scala.tools.nsc.Global.expandedDocComment(Global.scala:36)
>  [error] 	at scala.tools.nsc.doc.model.CommentFactory.defineComment(CommentFactory.scala:73)
>  [error] 	at scala.tools.nsc.doc.model.CommentFactory.defineComment$(CommentFactory.scala:38)
>  [error] 	at scala.tools.nsc.doc.DocFactory$$anon$1.defineComment(DocFactory.scala:68)
>  [error] 	at scala.tools.nsc.doc.model.CommentFactory.$anonfun$comment$1(CommentFactory.scala:31)
>  [error] 	at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:82)
>  [error] 	at scala.tools.nsc.doc.model.CommentFactory.comment(CommentFactory.scala:31)
>  [error] 	at scala.tools.nsc.doc.model.CommentFactory.comment$(CommentFactory.scala:29)
>  [error] 	at scala.tools.nsc.doc.DocFactory$$anon$1.comment(DocFactory.scala:68)
>  [error] 	at scala.tools.nsc.doc.model.ModelFactory$NonTemplateMemberImpl.$anonfun$comment$4(ModelFactory.scala:489)
>  [error] 	at scala.tools.nsc.doc.model.ModelFactory$NonTemplateMemberImpl.comment$lzycompute(ModelFactory.scala:489)
>  [error] 	at scala.tools.nsc.doc.model.ModelFactory$NonTemplateMemberImpl.comment(ModelFactory.scala:475)
>  [error] 	at scala.tools.nsc.doc.model.ModelFactory$MemberImpl.deprecation(ModelFactory.scala:160)
>  [error] 	at scala.tools.nsc.doc.model.Entity$.isDeprecated(Entity.scala:62)
>  [error] 	at scala.tools.nsc.doc.model.Entity$.$anonfun$EntityOrdering$1(Entity.scala:73)
>  [error] 	at scala.math.Ordering$$anon$5.compare(Ordering.scala:122)
>  [error] 	at scala.math.Ordering$$anon$5.compare(Ordering.scala:122)
>  [error] 	at java.util.TimSort.binarySort(TimSort.java:296)
>  [error] 	at java.util.TimSort.sort(TimSort.java:221)
>  [error] 	at java.util.Arrays.sort(Arrays.java:1438)
>  [error] 	at scala.collection.SeqLike.sorted(SeqLike.scala:656)
>  [error] 	at scala.collection.SeqLike.sorted$(SeqLike.scala:644)
>  [error] 	at scala.collection.AbstractSeq.sorted(Seq.scala:41)
>  [error] 	at scala.tools.nsc.doc.html.page.EntityPage.$init$(Entity.scala:167)
>  [error] 	at scala.tools.nsc.doc.html.page.EntityPage$$anon$1.<init>(Entity.scala:1117)
>  [error] 	at scala.tools.nsc.doc.html.page.EntityPage$.apply(Entity.scala:1117)
>  [error] 	at scala.tools.nsc.doc.html.HtmlFactory.$anonfun$writeTemplates$1$adapted(HtmlFactory.scala:129)
>  [error] 	at scala.tools.nsc.doc.html.HtmlFactory.writeTemplates(HtmlFactory.scala:131)
>  [error] 	at scala.tools.nsc.doc.html.HtmlFactory.generate(HtmlFactory.scala:116)
>  [error] 	at scala.tools.nsc.doc.html.Doclet.generateImpl(Doclet.scala:23)
>  [error] 	at scala.tools.nsc.doc.doclet.Generator.generate(Generator.scala:23)
>  [error] 	at scala.tools.nsc.doc.DocFactory.generate$1(DocFactory.scala:128)
>  [error] 	at scala.tools.nsc.doc.DocFactory.document(DocFactory.scala:131)
>  [error] 	at xsbt.Runner.run(ScaladocInterface.scala:30)
>  [error] 	at xsbt.ScaladocInterface.run(ScaladocInterface.scala:15)
>  [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>  [error] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>  [error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>  [error] 	at java.lang.reflect.Method.invoke(Method.java:498)
>  [error] 	at sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:237)
>  [error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:166)
>  [error] 	at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:147)
>  [error] 	at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:36)
>  [error] 	at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:36)
>  [error] 	at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:78)
>  [error] 	at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:71)
>  [error] 	at sbt.RawCompileLike$.$anonfun$cached$3(RawCompileLike.scala:63)
>  [error] 	at sbt.RawCompileLike$.$anonfun$cached$3$adapted(RawCompileLike.scala:61)
>  [error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
>  [error] 	at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:68)
>  [error] 	at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:52)
>  [error] 	at sbt.Defaults$.$anonfun$docTaskSettings$3(Defaults.scala:1324)
>  [error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
>  [error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:39)
>  [error] 	at sbt.std.Transform$$anon$4.work(System.scala:66)
>  [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:263)
>  [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
>  [error] 	at sbt.Execute.work(Execute.scala:272)
>  [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:263)
>  [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:174)
>  [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
>  [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
>  [error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
>  [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
>  [error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
>  [error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
>  [error] 	at java.lang.Thread.run(Thread.java:748)
>  [error] (Compile / doc) java.lang.UnsupportedOperationException: Position.start on NoPosition
>  [error] Total time: 11 s, completed 22-Aug-2018 12:27:04
